modules/internal/ChapelRange.chpl

```diff
@@ -861,7 +862,12 @@ proc _cast(type t, r: range(?)) where isRangeType(t) {
 
     r1._low = r2._low;
     r1._high = r2._high;
-    r1._stride = r2._stride;
+
+    if s1 && s2 then
+      r1._stride = r2.stride;
+    else if s1 then
+      r1._stride = 1;
```

Comment:
Nice!  Integrating the !s1 && s2 compilerError above with this control flow might help readers like me who found themselves asking "What about the missing s2 && !s1 case...?" before noticing it's handled separately and earlier.